src/Util/utils.php

```diff
@@ -0,0 +1,201 @@
+                $item;
+            $index++;
+            return $list;
+        }, '');
+}
+
+/**
+ * Given an invalid input string and a list of valid options, returns a filtered
+ * list of valid options sorted based on their similarity with the input.
+ *
+ * @param string $input
+ * @param array  $options
+ * @return array
+ */
+function suggestionList(string $input, array $options): array
+{
+    $optionsByDistance = [];
+    $oLength = \count($options);
+    $inputThreshold = \strlen($input) / 2;
+
+    /** @noinspection ForeachInvariantsInspection */
+    for ($i = 0; $i < $oLength; $i++) {
+        // Comparison must be case-insenstive.
+        $distance = \levenshtein(\strtolower($input), \strtolower($options[$i]));
+        $threshold = \max($inputThreshold, \strlen($options[$i]) / 2, 1);
+        if ($distance <= $threshold) {
+            $optionsByDistance[$options[$i]] = $distance;
+        }
+    }
+
+    $result = \array_keys($optionsByDistance);
+
+    \usort($result, function ($a, $b) use ($optionsByDistance) {
+        return $optionsByDistance[$a] - $optionsByDistance[$b];
+    });
+
+    return $result;
+}
+
+/**
+ * Given `[A, B, C]` returns `'"A", "B" or "C"'`.
+ *
+ * @param array $items
+ * @return string
+ */
+function quotedOrList(array $items): string
+{
+    return orList(array_map(function ($item) {
+        return '"' . $item . '"';
+    }, $items));
+}
+
+
+
+/**
+ * @param array    $array
+ * @param callable $fn
+ * @return bool
+ */
+function arrayEvery(array $array, callable $fn): bool
+{
+    return array_reduce($array, function ($result, $value) use ($fn) {
+        return $result && $fn($value);
+    }, true);
+}
+
+/**
+ * @param array    $array
+ * @param callable $fn
+ * @return mixed
+ */
+function arraySome(array $array, callable $fn)
+{
+    return array_reduce($array, function ($result, $value) use ($fn) {
+        return $result || $fn($value);
+    });
+}
+
+/**
+ * @param array    $array
+ * @param callable $predicate
+ * @return mixed|null
+ */
+function find(array $array, callable $predicate)
+{
+    foreach ($array as $value) {
+        if ($predicate($value)) {
+            return $value;
+        }
+    }
+
+    return null;
+}
+
+/**
+ * @param array    $array
+ * @param callable $keyFn
+ * @return array
+ */
+function keyMap(array $array, callable $keyFn): array
+{
+    return array_reduce($array, function ($map, $item) use ($keyFn) {
+        $map[$keyFn($item)] = $item;
+        return $map;
+    }, []);
+}
+
+/**
+ * @param array    $array
+ * @param callable $keyFn
+ * @param callable $valFn
+ * @return array
+ */
+function keyValueMap(array $array, callable $keyFn, callable $valFn): array
+{
+    return array_reduce($array, function ($map, $item) use ($keyFn, $valFn) {
+        $map[$keyFn($item)] = $valFn($item);
+        return $map;
+    }, []);
+}
+
+/**
+ * @param $value
+ * @return string
+ */
+function toString($value): string
```

Comment:
I just wonder can we rename it to `getType` or something like that? `toString` doesn't reflect its function

Reply:
It does, it works for any kind of value; objects, array, scalars, null, etc.

Reply:
But normally, I expects to get a string value of given variable, instead of its type as string.